### ISSUE TITLE
Map moves out of view when resizing

### ISSUE THREAD

**Issue by Lexikus** [amcharts/amcharts4#2684]
There is an issue when a `MapChart`  is used in combination with a `MapPolygonSeries` when data needs to be lazyloaded from the network. When you resize the HTML container in the jsFiddle, the maps moves out of the view. Don't move the map first around.

The code below can be put in jsFiddle to see the issue.

```html
<script src="https://www.amcharts.com/lib/4/core.js"></script>
<script src="https://www.amcharts.com/lib/4/maps.js"></script>
<script src="https://www.amcharts.com/lib/4/geodata/worldHigh.js"></script>
<script src="https://www.amcharts.com/lib/4/geodata/data/countries2.js"></script>
<script src="https://www.amcharts.com/lib/4/themes/animated.js"></script>
<div id="chartdiv" style="width: 100%; height: 400px"></div>
```

```javascript
const chart = am4core.create('chartdiv', am4maps.MapChart);
chart.projection = new am4maps.projections.Miller();

const worldSeries = chart.series.push(new am4maps.MapPolygonSeries());
worldSeries.useGeodata = true;

worldSeries.heatRules.push({
    property: 'fill',
    target: worldSeries.mapPolygons.template,
    min: chart.colors.getIndex(1).brighten(1),
    max: chart.colors.getIndex(1).brighten(-0.3)
});

const worldPolygon = worldSeries.mapPolygons.template;
worldPolygon.tooltipText = '{name}: {value}';
worldPolygon.nonScalingStroke = true;
worldPolygon.strokeOpacity = 0.5;
worldPolygon.fill = am4core.color('#aaa');
worldPolygon.propertyFields.fill = 'color';

const worldPolygonHoverState = worldPolygon.states.create('hover');
worldPolygonHoverState.properties.fill = chart.colors.getIndex(0);

const heatLegend = chart.createChild(am4maps.HeatLegend);
heatLegend.series = worldSeries;
heatLegend.align = 'right';
heatLegend.valign = 'bottom';
heatLegend.width = am4core.percent(20);
heatLegend.marginRight = am4core.percent(4);
heatLegend.minValue = 0;
heatLegend.maxValue = 100;
heatLegend.valueAxis.renderer.labels.template.adapter.add('text', labelText => '');

const minRange = heatLegend.valueAxis.axisRanges.create();
minRange.value = heatLegend.minValue;
const maxRange = heatLegend.valueAxis.axisRanges.create();
maxRange.value = heatLegend.maxValue;

worldSeries.geodataSource.events.on('parseended', ev => {
  const countrySeriesData = [];
  for (const region of ev.target.data.features) {
    const data = {
      id: region.id,
      value: 1
    };
    countrySeriesData.push(data);
  }

  worldSeries.data = countrySeriesData;
});

worldSeries.geodataSource.url = 'https://www.amcharts.com/lib/4/geodata/json/switzerlandLow.json';
worldSeries.geodataSource.load();
```

I found a workaround to fix this problem by changing the geodata source holder to the parent of the series.

Basically, I changed this and it works better:

```javascript
// from:
worldSeries.geodataSource.events.on('parseended', ev => {
  // ...
  worldSeries.data = countrySeriesData;
});

worldSeries.geodataSource.url = 'https://www.amcharts.com/lib/4/geodata/json/switzerlandLow.json';
worldSeries.geodataSource.load();

// to
chart.geodataSource.events.on('parseended', ev => {
  // ...
  worldSeries.data = countrySeriesData;
});

chart.geodataSource.url = 'https://www.amcharts.com/lib/4/geodata/json/switzerlandLow.json';
```

Also, the heat legend seems to be off on first load. After resizing it is fine.

**Comment by martynasma** [comment]
Fixed in [4.9.29](https://github.com/amcharts/amcharts4/releases/tag/4.9.29).

## [4.9.29] - 2020-07-09

### Changed
- Export: data exported as HTML will now have `<thead>` and `<tbody>` tags.
- `XYChart.getClosest()` method is now public.
- Accessibility: Now pressing ENTER while there's a `MapChart` element focused (using TAB key) will invoke its `hit` event, if set.

### Fixed
- JSON: heat rules referring to series' bullets were not working, e.g. `target: "bullet[0].circle"`.
- IE: More fixes to polyfill overriding.
- `ColorSet` method `next()` would fail if `currentStep` was bigger than `minColors`, affecting `SliceGrouper` plugin setups with a lot of slices in particular.
- On `MapChart` settings `minZoomLevel` and `maxZoomLevel` were being ignored when pinch-zooming.
- Syncing of axes improved, especially when series of one axis were hidden/shown.
- If a `MapSeries` had its geodata loaded via `geodataSource`, the map used to disappear after div size changed.
- Legend had a `maxWidth` set to `200` and this could result labels to be truncated even if there was enough space for them to fit in.
- Ticks/grid could disappear if `ValueAxis` had `min`/`max` set and data of a chart changed.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by saveriocaminitiktech** [comment]
I think this issue is still partially there.
I'm experiencing this "map moves out of view on resize"-effect when a map loaded from an external json is used in conjunction with an instance of MapPolygonSeries.
I created a minimal viable example that consistently reproduce the bug for me on MacOS:
https://codepen.io/saveriocaminitiktech/pen/bGMVmyV
The JS code is properly commented for your convenience.

Observing that the bug goes away if the user initially drag the map,
I realised that calling chart.goHome(0) on chart ready event can be a valid workaround.
Enable the workaround code line at the end of the provided codepen to check this out.

**Comment by earrechecds** [comment]
It is still happening sadly on amcharts5, the goHome didn't work to fix it either, any suggestions?
Steps to reproduce:
- see the map as a globe
- go to antartida, zoom in and out with the wheel
the globe gets cut, and the spin of the globe is strange after too. 
it happens with all the countries more or less
